Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -0,0 +1,411 @@
+
+  btns_layout->addWidget(m_exclude_size_label, 1, 0);
+  btns_layout->addWidget(m_include_size_label, 1, 1);
+
+  m_matching_results_list = new QListWidget();
+  m_matching_results_list->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  m_matching_results_list->setContextMenuPolicy(Qt::CustomContextMenu);
+  m_reset_btn = new QPushButton(tr("Reset All"));
+  m_reset_btn->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+
+  auto* layout = new QVBoxLayout();
+  layout->addLayout(btns_layout);
+  layout->addLayout(labels_layout);
+  layout->addWidget(m_matching_results_list);
+  layout->addWidget(m_reset_btn);
+
+  setLayout(layout);
+}
+
+void CodeDiffDialog::ConnectWidgets()
+{
+  connect(m_record_btn, &QPushButton::toggled, this, &CodeDiffDialog::OnRecord);
+  connect(m_include_btn, &QPushButton::pressed, [this]() { Update(true); });
+  connect(m_exclude_btn, &QPushButton::pressed, [this]() { Update(false); });
+  connect(m_matching_results_list, &QListWidget::itemClicked, [this](QListWidgetItem* item) {
+    m_code_widget->SetAddress(item->data(Qt::UserRole).toUInt(),
+                              CodeViewWidget::SetAddressUpdate::WithUpdate);
+  });
+  connect(m_reset_btn, &QPushButton::pressed, this, &CodeDiffDialog::ClearData);
+  connect(m_matching_results_list, &CodeDiffDialog::customContextMenuRequested, this,
+          &CodeDiffDialog::OnContextMenu);
+}
+
+void CodeDiffDialog::ClearData()
+{
+  if (m_record_btn->isChecked())
+    m_record_btn->toggle();
+  ClearBlockCache();
+  m_matching_results_list->clear();
+  m_exclude_size_label->setText(QStringLiteral("Excluded: 0"));
+  m_include_size_label->setText(QStringLiteral("Included: 0"));
+  m_exclude_btn->setEnabled(false);
+  m_include_btn->setEnabled(false);
+  std::vector<Diff>().swap(m_include);
+  std::vector<Diff>().swap(m_exclude);
+  JitInterface::SetProfilingState(JitInterface::ProfilingState::Disabled);
+}
+
+void CodeDiffDialog::ClearBlockCache()
+{
+  Core::State old_state = Core::GetState();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Paused);
+
+  JitInterface::ClearCache();
+
+  if (old_state == Core::State::Running)
+    Core::SetState(Core::State::Running);
+}
+
+void CodeDiffDialog::OnRecord(bool enabled)
+{
+  JitInterface::ProfilingState state;
+
+  if (enabled)
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Recording..."));
+    state = JitInterface::ProfilingState::Enabled;
+    m_exclude_btn->setEnabled(true);
+    m_include_btn->setEnabled(true);
+  }
+  else
+  {
+    ClearBlockCache();
+    m_record_btn->setText(tr("Start Recording"));
+    state = JitInterface::ProfilingState::Disabled;
+    m_exclude_btn->setEnabled(false);
+    m_include_btn->setEnabled(false);
+  }
+
+  m_record_btn->update();
+  JitInterface::SetProfilingState(state);
+}
+
+void CodeDiffDialog::OnInclude()
+{
+  std::vector<Diff> current_diff;
+  auto current_symbols = CalculateSymbolsFromProfile();

```

Comment:
Can't it be marked as `const auto current_symbols`? This should probably be renamed as `included_symbols` or similar.

Reply:
Changed to const.
As for renaming current_symbols to included_symbols / excluded_symbols I think its kind of misleading.

OnInclude/OnExclude() is the operation taking place after a recording. CalculateSymbolsFromProfile is getting the recorded symbols since the profile/recording ended. 
Then checks are done with the lists and current_diff holds the real result.

So current_symbols is not only the included or excluded symbols.

For clarity I changed current_symbols in those two functions to `recorded_symbols`

Reply:
Indeed, this naming is more fitting. It doesn't look like you've pushed your changes, though.

Reply:
Double checked this is changed, resolving.